packages/core/src/components/region/RegionSlider/RegionSlider.tsx

```diff
@@ -204,17 +210,52 @@ function RegionSlider() {
   }
 
   // The `shouldClearPickupPointsSimulation` param prevent triggering store update more than once.
-  const onDismissSlider = ({
+  const onDismissSlider = async ({
     shouldClearPickupPointsSimulation = true,
   }: { shouldClearPickupPointsSimulation?: boolean } = {}) => {
-    setDataLoading(false)
+    setDataLoading(true)
     setInput(session.postalCode)
     setAppliedInput(session.postalCode)
     setValidatedSession(undefined)
-    setPickupPointOption(undefined)
+    setPickupPointOption(null)
     shouldClearPickupPointsSimulation && clearPickupPointsSimulation()
   }
 
+  const clearFilter = async () => {
+    const selectedPickupInPointFacets = searchState.selectedFacets.filter(
+      ({ key, value }) =>
+        value === PICKUP_IN_POINT_FACET_VALUE || key === PICKUP_POINT_FACET_KEY
+    )
+
+    if (selectedPickupInPointFacets.length === 0) {
```

Comment:
I forgot, this is the scenario when no facets selected, but we just need to clear the state right?

Reply:
Yes, basically the shopper is selecting for the first time and canceled the action, so we just need to clear the `RegionSlider` state